### PR TITLE
Add support for ruff

### LINKED ISSUE #17: Add support for ruff
https://github.com/astral-sh/ruff

It would be nice to add `vcs-diff-lint-csdiff-ruff` wrapper or a `RuffLinter` class into `vcs-diff-lint`.

### PR DESCRIPTION
**Pull request by FrostyX**
Fix #17
See https://docs.astral.sh/ruff

### CODE REVIEW

File: vcs-diff-lint-csdiff-ruff [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import sys
+import json
+from subprocess import Popen, PIPE
+
+
+def ruff_check():
+    """
+    Run `ruff check` and return a dict with its results
+    """
+    cmd = ["ruff", "check", "--output-format=json"] + sys.argv[1:]
```
Comment (xsuchy) [inline vcs-diff-lint-csdiff-ruff:21]: Should it be added as a weak deps to spec file?

Should the argv be sanitized?

Comment (FrostyX) [reply vcs-diff-lint-csdiff-ruff:21]: > Should it be added as a weak deps to spec file?

It should, thank you. Good catch.

> Should the argv be sanitized?

I don't know, I did the same that `vcs-diff-lint-csdiff-pylint` does. But probably not? Worst case scenario, the bad values are going to be passed to `ruff check` which will handle it and error out.

Comment (xsuchy) [reply vcs-diff-lint-csdiff-ruff:21]: Right, with the shell=False (that is the default) this is not a problem.

**Review by xsuchy — APPROVED** [review]
+1

**Review by praiskup — APPROVED** [review]

**Comment by praiskup** [comment]
Thank you!

**Pull request merged**